eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml

```diff
@@ -1,18 +1,14 @@
+parameters:
+  ScriptDirectory: 'eng/common/scripts/git-diff-changes.ps1'
 steps:
   - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
     - pwsh: |
         Set-PsDebug -Trace 1
         # Find the default branch of the repo
-        $defaultBranch = (git remote show origin | Out-String) -replace "(?ms).*HEAD branch: (\w+).*", '$1'
-        $targetBranch = "$(System.PullRequest.TargetBranch)" -replace "refs/heads/"
-
-        Write-Host "Default Branch: ${defaultBranch}"
-
-        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and $targetBranch -eq $defaultBranch)
```

Comment:
What's the reason for removing this part of the logic?

Reply:
It should add back. Good catch

Reply:
Could you help me understand what situation are we checking here?
`$targetBranch -eq $defaultBranch`

Reply:
Basically, there is not need to have this check if the PR is not being merged into main. So this just checks if the targetBranch is main. The reason for not using `main` as a literal is because of the general guidance of not hardcoding the branch name. Remember the master to main work fromlast year.